### DISCUSSION TITLE
基于 ChatGPT 的 epub 书籍自动翻译工具

### DISCUSSION THREAD

**Discussion by will-ww** [X-lab2017/open-wonderland#256] [1️⃣ 新鲜事]
开源项目地址：https://github.com/yihong0618/bilingual_book_maker

翻译 epub 书籍示例：

[Coding_Freedom_中英对照.zip](https://github.com/X-lab2017/open-wonderland/files/10889886/Coding_Freedom_.zip)

来源：庄老师~

**Comment by yoyo-wu98** [comment]
挺好玩的，不过看了代码，主要的部分是向 chatgpt 写入 "Please help me to translate,`{text}` to {self.language}, please return only translated content not include the origin text"

需要担心，这种翻译方式是否可以达成上下文的交互和内涵传播（这一点其实是chatgpt最擅长的部分），建议改进这个prompt，比如加上”based on the previous context“ 或者先过一遍文章再整体翻译（输出的时候依然是一段一段的，但是翻译的时候是以一个整体，这是最重要的一点）
